velox/functions/lib/Utf8Utils.h

```diff
@@ -48,4 +50,32 @@ namespace facebook::velox::functions {
 /// https://github.com/airlift/slice/blob/master/src/main/java/io/airlift/slice/SliceUtf8.java
 int32_t tryGetCharLength(const char* input, int64_t size);
 
+/// Return the length in byte of the next UTF-8 encoded character at the
+/// beginning of `string`. If the beginning of `string` is not valid UTF-8
+/// encoding, return -1.
+FOLLY_ALWAYS_INLINE int validateAndGetNextUtf8Length(
+    const unsigned char* string,
+    const unsigned char* end) {
+  if (string == end) {
```

Comment:
We can remove this branch (and replace it with a `VELOX_DCHECK_LT`) since it's guaranteed on all our call sites

Reply:
Make sense! Revised